File: lab9/sha.py
import hashlib
def sha256(input):
    # Create a new SHA-256 hash object
    hash = hashlib.sha256()
    # Update the hash object with the input string
    hash.update(input.encode('utf-8'))
    # Get the hex digest of the hash value
    output = hash.hexdigest()
    return output
    # Test the function with a sample input
input = "Hello, world!"
output = sha256(input)
print("Hash Value:", output)

File: lab9/sha.cpp
#include <iostream>
#include <iomanip>
#include <string>
#include <cstring>
#include <sstream>
#include <openssl/sha.h>

using namespace std;

string sha256(string input) {
    unsigned char digest[SHA256_DIGEST_LENGTH];
    SHA256((const unsigned char *)input.c_str(), input.length(), digest);

    stringstream ss;
    for(int i = 0; i < SHA256_DIGEST_LENGTH; i++) {
        ss << hex << setw(2) << setfill('0') << (int)digest[i];
    }

    return ss.str();
}

int main() {
    string input = "Hello, world!";
    string output = sha256(input);
    cout <<"Hash value using SHA256: "<<output << endl;

    return 0;
}
